crates/sui-json/src/lib.rs

```diff
@@ -818,6 +819,27 @@ fn resolve_call_arg(
     }
 }
 
+pub fn is_receiving_argument(view: &BinaryIndexedView, arg_type: &SignatureToken) -> bool {
+    let mut current_token = Some(arg_type);
+    while let Some(sig_tok) = current_token {
+        match sig_tok {
+            // Progress down into references to determine if the underlying type is a receiving
+            // type or not.
+            SignatureToken::Reference(inner) | SignatureToken::MutableReference(inner) => {
+                current_token = Some(inner);
+            }
+            SignatureToken::StructInstantiation(sidx, targs) => {
+                return resolve_struct(view, *sidx) == RESOLVED_RECEIVING_STRUCT
+                    && targs.len() == 1;
+            }
+            _ => {
+                return false;
+            }
+        }
+    }
+    false
```

Comment:
Do you need to use an `Option` for `current_token`?

```suggestion
    let mut current_token = arg_type;
    loop {
        match current_token {
            // Progress down into references to determine if the underlying type is a receiving
            // type or not.
            SignatureToken::Reference(inner) | SignatureToken::MutableReference(inner) => {
                current_token = inner,
            }
            SignatureToken::StructInstantiation(sidx, targs) => {
                return resolve_struct(view, *sidx) == RESOLVED_RECEIVING_STRUCT
                    && targs.len() == 1;
            }
            _ => {
                return false;
            }
        }
    }
    false
```

Or alternatively:

```suggestion
    let token = arg_type;
    use SignatureToken as ST;
    while let (ST::Reference(inner) | ST::MutableReference(inner)) = token {
        token = inner;
    }
    
    matches!(
        token, 
        ST::StructInstantiation(sidx, [_]) 
        if resolve_struct(view, *sidx) == RESOLVED_RECEIVING_STRUCT 
    )
```

Reply:
I love the last one! I will definitely make this change, and thanks for the suggestion!